java/src/com/ibm/streamsx/topology/function/FunctionContext.java

```diff
@@ -0,0 +1,145 @@
+package com.ibm.streamsx.topology.function;
+
+import java.net.MalformedURLException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+
+import com.ibm.streams.operator.ProcessingElement;
+
+/**
+ * Context for a function executing in a IBM Streams application.
+ */
+public interface FunctionContext {
+    
+    /**
+     * Get the processing element hosting a function 
+     * @return ProcessingElement hosting a function.
+     */
+    ProcessingElement getPE();
```

Comment:
Hadn't really looked all that closely at it.  It was more out of a general observation.

Does the Java Operator API "leak through" elsewhere other than when explicitly using something in com.ibm.streamsx.topology.spl?  The public api seems overwhelmingly independent of the Java Operator API and doing something here seems like a small price to pay to keep it that way.

Unrelated to whether or not we return that interface, it seems like additional doc is needed somewhere to describe certain values under certain execution contexts - i.e., for StreamContext.STANDALONE / when ProcessingElement.isStandalone()==true, the behavior of getJobId(), getPEId(), getInstanceId(), getDomainId()